tests/Unit/Css/CssDocumentTest.php

```diff
@@ -488,6 +523,25 @@ public function notRendersInvalidNonConditionalAtRule(string $atRuleCss, string
         self::assertStringNotContainsString($atAndRuleName, $result);
     }
 
+    /**
+     * @test
+     *
+     * @dataProvider provideInvalidNonConditionalAtRule
+     */
+    public function throwsExceptionForInvalidNonConditionalAtRuleInDebugMode(
+        string $atRuleCss,
+        string $cssBefore = ''
+    ): void {
+        $this->expectException(UnexpectedTokenException::class);
+
+        $this->createDebugSubject($cssBefore . $atRuleCss);
+
+        self::markTestSkipped(
```

Comment:
Please let's have the `markTestSkipped` calls at the start of the test so no other code gets run. (This also helps the reader.)

Reply:
That would skip the test for all datasets, including the ones that currently pass.

The test data consists of CSS that contains some invalid declarations.  We test to confirm that in non-debug mode the invalid CSS is dropped.  In debug ("strict") mode, the CSS parser throws exceptions for some but not all invalid CSS constructs.  So we cannot assert that an exception is thrown for all the CSS strings in the data.

But if we don't assert it for any, we may as well not bother with the test at all, or have a dummy test that does not use the data provider but calls `markTestSkipped()` so we just see one skipped test in the results (not one for each dataset).

An alternative would be to split the datasets (data providers) up so that one provides datasets for which the CSS parser currently throws an exception, and one provides those for which it doesn't.  Though there doesn't appear to be any particular distinctive feature of those for which an exception is thrown - it's just cases where the strictness isn't quite as strict as the spec dictates.

WDYT?

Reply:
> An alternative would be to split the datasets (data providers) up so that one provides datasets for which the CSS parser currently throws an exception, and one provides those for which it doesn't.

That's the approach I'd prefer.

Reply:
OK.  I've split up the data accordingly, only passing the CSS for which the CSS parser throws an exception to the tests for that, with a comment in the DocBlock indicating that the other CSS is not currently tested.